lib/setup.js

```diff
@@ -437,6 +437,17 @@ function processCommand(command, args, params, callback) {
             // If user accidentally wrote tools.appName.adapter => remove adapter
             name = cli.tools.normalizeAdapterName(name);
 
+            if (!cli.tools.validateAdapterOrInstanceIdentifier(name)) {
+                showHelp();
+                return void callback(2);
+            }
+
+            // split the adapter into adapter + instance if necessary
+            if (name.includes('.')) {
+                ([name, instance] = name.split('.', 2));
+                params.instance = instance;
+            }
```

Comment:
Shouldn't we first check if instance is numeric?
This way we could write `iobroker add admin.ffff` and `instance` would be set to `ffff`, which would probably break other stuff.

Reply:
No, see https://github.com/ioBroker/ioBroker.js-controller/pull/923/files#diff-f5d8c0f33aa88aa6e119a0cdd5f1c40eR440